GoogleChromeLabs/ps-analysis-tool
# Refactor: Cookie filtering logic

## Description
The filtering logic for the cookie table data had some bugs where only the latest filter would apply. This PR aims to remove that bug and provide correct data post-filtering.

## Relevant Technical Choices

The filtering function has been split into 3 parts:
- A function which filters the cookie-based only on `retentionPeriod`.
- A function which filters the cookie based on all other filters except `rententionPeriod`.
- A function which controls all the possible combinations of the filters (i.e `retentionPeriod` and other filters case).

## Testing Instructions
- Pull this branch.
- In the terminal run 'npm i && npm run dev`.
- Open the browser got to `indianexpress.com`.
- Open the DevTools and click on privacy sandbox.
- Open the cookie dropdown and select any frame which has cookies.
- Clicking on the filtering Icon should open the filter UI.
- Toggling the filtering UI should give you the correct data in the cookie table.

## Screenshot/Screencast


https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/17cb9f13-304d-4282-8fed-73cdd2257a2d
